mlflow/gateway/providers/openai.py

```diff
@@ -159,6 +162,43 @@ async def completions(self, payload: completions.RequestPayload) -> completions.
         # The range of OpenAI's temperature is 0-2, but ours is 0-1, so we double it.
         payload["temperature"] = 2 * payload["temperature"]
         payload["messages"] = [{"role": "user", "content": payload.pop("prompt")}]
+        return payload
+
+    def _prepare_completion_response_payload(self, resp):
+        return completions.ResponsePayload(
+            **{
+                "candidates": [
+                    {
+                        "text": c["message"]["content"],
+                        "metadata": {"finish_reason": c["finish_reason"]},
+                    }
+                    for c in resp["choices"]
+                ],
+                "metadata": {
+                    "input_tokens": resp["usage"]["prompt_tokens"],
+                    "output_tokens": resp["usage"]["completion_tokens"],
+                    "total_tokens": resp["usage"]["total_tokens"],
+                    "model": resp["model"],
+                    "route_type": self.config.route_type,
+                },
+            }
+        )
+
+    def sync_completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
```

Comment:
Is there a reason we need this sync? can't the consumer of this call await the request?

Reply:
https://github.com/mlflow/mlflow/pull/9538#discussion_r1318115811. asyncio doesnt work in databricks notebook

Reply:
:'( 

This is sad. But thanks for the information